ml-agents/mlagents/trainers/settings.py

```diff
@@ -464,8 +596,7 @@ class RunOptions(ExportableSettings):
     )
     env_settings: EnvironmentSettings = attr.ib(factory=EnvironmentSettings)
     engine_settings: EngineSettings = attr.ib(factory=EngineSettings)
-    parameter_randomization: Optional[Dict[str, ParameterRandomizationSettings]] = None
-    curriculum: Optional[Dict[str, CurriculumSettings]] = None
+    environment_parameters: Optional[Dict[str, EnvironmentParameterSettings]] = None
```

Comment:
I think what we can do here is rather than have the `EnvironmentParameterSettings` class handle both cases, set this type as a Union:
```suggestion
    environment_parameters: Optional[Dict[str, Union[ParameterRandomizationSettings, EnvironmentParameterSettings]]] = None
```
`cattr` during the structure will auto-detect whether it's an EnvironmentParameterSettings or ParameterRandomizationSettings based on the keys in the dict. In `EnvironmentParamterSettings.structure` you then don't need to worry about the `ParameterRandomizationSettings` case, that can stay in `ParameterRandomizationSettings`.

Reply:
I do think it would be cleaner to handle everything in EnvironmentParamterSettings.structure rather than maintaining the conversion for both ParameterRandomizationSettings and EnvironmentParamterSettings

Reply:
IMO the latter is more modular, e.g. if we were to use ParameterRandomizationSettings in a different way as part of another Settings class we could reuse that method. But not super adamant about it if keeping it together is cleaner.

Reply:
Perhaps I'm late to the party but as it is, the implementation doesn't preclude the use of a ParameterRandomizationSettings as part of another settings class since the class itself still has a structure function.

Reply:
Yeah I think the current implementation is quite different than the original, which didn't use the ParameterRandomizationSettings structure method.